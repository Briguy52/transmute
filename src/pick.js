import _filter from './internal/_filter';
import curry from './curry';
import { Seq } from 'immutable';

function pick(keys, subject) {
  const keySet = Seq.Set(keys);
  return _filter((value, key) => keySet.contains(key), subject);
}

/**
 * Select specified keys from a KeyedIterable (e.g. a `Map` or `OrderedMap`).
 *
 * @example
 * // returns Map { "one" => 1, "three" => 3 }
 * pick(
 *   ['one', 'three'],
 *   Map({one: 1, two: 2, three: 3})
 * );
 *
 * @param  {Array|Iterable|Object} keys to select.
 * @param  {KeyedIterable} subject from which to select `keys`.
 * @return {KeyedIterable} with just `keys`.
 */
export default curry(pick);
