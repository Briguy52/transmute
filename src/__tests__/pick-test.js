import { Map, Record } from "immutable";
import pick from "../pick";

describe("transmute/pick", () => {
  const takeOneAndThree = pick(["one", "three"]);

  it("selects keys from Maps", () => {
    expect(
      takeOneAndThree(Map({ one: 1, two: 2, three: 3 }))
    ).toMatchSnapshot();
  });

  it("selects keys from Object", () => {
    const testObject = {
      one: 1,
      two: 2,
      three: 3
    };
    expect(takeOneAndThree(testObject)).toMatchSnapshot();
  });

  it("selects keys from Records", () => {
    const TestRecord = Record(
      {
        one: 1,
        two: 2,
        three: 3
      },
      "TestRecord"
    );
    expect(takeOneAndThree(TestRecord())).toMatchSnapshot();
  });
});
