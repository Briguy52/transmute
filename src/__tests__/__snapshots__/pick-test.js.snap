// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/pick selects keys from Maps 1`] = `
Immutable.Map {
  one: 1,
  three: 3,
}
`;

exports[`transmute/pick selects keys from Object 1`] = `
Object {
  "one": 1,
  "three": 3,
}
`;

exports[`transmute/pick selects keys from Records 1`] = `
Immutable.Map {
  one: 1,
  three: 3,
}
`;
